zulipterminal/ui_tools/boxes.py

```diff
@@ -131,11 +120,28 @@ def stream_box_view(self, stream_id: int, caption: str='', title: str='',
         ]
         self.contents = write_box
 
+    def stream_box_view(self, stream_id: int, caption: str='', title: str='',
+                        ) -> None:
+        self.stream_write_box = ReadlineEdit(
+            caption="Stream:  ",
+            edit_text=caption
+        )
+        self.stream_write_box.enable_autocomplete(
+            func=self._stream_box_autocomplete,
+            key=primary_key_for_command('AUTOCOMPLETE'),
+            key_reverse=primary_key_for_command('AUTOCOMPLETE_REVERSE')
+        )
+        self.stream_write_box.set_completer_delims("")
+        self._stream_view(stream_id=stream_id, title=title)
+
     def stream_box_edit_view(self, stream_id: int, caption: str='',
                              title: str='') -> None:
-        self.stream_box_view(stream_id, caption, title)
+        self.in_edit_message_box = True
+        self.edit_caption = caption
+        self.stream_write_box = urwid.AttrMap(urwid.SelectableIcon(
+                                "Stream: " + caption), None, 'selected')
+        self._stream_view(stream_id=stream_id, title=title)
```

Comment:
Since the wrapper function's sole purpose lies in setting the box in edit mode, we could redeclare `self.stream_write_box` in the same :)
Also, it would be cleaner if you add the new method `self._stream_view` below the old `self.stream_box_view` instead of renaming it, so as to get rid of the shifted diffs having the same code :)

Reply:
We basically have to modify the widget list in that case, but that would lead to more duplication as we have to edit the `header_write_box ` and then finally `write_box`, which basically would mean half the `stream_write_box` function itself. I preferred the current approach as it leads to the least duplication.
The f string one I definitely agree with.

Reply:
Ah, I misled my above comment. Actually, I was talking about setting the `self.header_write_box` widget list element (stream box) as `Urwid.Text()`, in edit mode. I think this should be straightforward? (i.e., setting this in `self.stream_box_edit_view()`)

Reply:
> Ah, I misled my above comment. Actually, I was talking about setting the `self.header_write_box` widget list element (stream box) as `Urwid.Text()`, in edit mode. I think this should be straightforward? (i.e., setting this in `self.stream_box_edit_view()`)

Any update on header_write_box would trigger the same changes as above.